Makefile

```diff
@@ -269,9 +271,9 @@ format: modtidy
 ################################################################################
 # Target: check                                                              #
 ################################################################################
-# TODO add lint
 .PHONY: check
-check: format test
+check: format test lint
+	git status && [[ -z `git status -s` ]]
```

Comment:
~~~
git status && [[ -z `git status -s` ]]
~~~
It will exit with code 1 when there is something untracked or uncommitted. 
It's better to exit with code 1 when something unexpected happens, so that we can reuse this script in other tools.
But if we modify it to:
```
[[ -z git status -s ]] || git status
```
It will always exit with code 0.